### PR TITLE
feat: Block scorable and state is persisted

### PR DESCRIPTION
**Pull request by ziafazal**
This PR has changes to
1. Make instruqt track scorable based on completed and total challenges.
2. Persist user sessions which mean return users start track where they left

### CODE REVIEW

File: instruqtxblock/instruqtxblock.py [hunk 1]
```diff
@@ -54,13 +55,51 @@ class InstruqtXBlock(StudioEditableXBlockMixin, CompletableXBlockMixin, XBlock):
...
 
+    def _get_user_id(self):
+        """
+        Returns external user id for lti type when runtime is LMS or returns `student_1`\
+             for other runtimes like workbench
+        """
+        user_service = self.runtime.service(self, 'user')
+        user_id = 'student_1'
+        if hasattr(user_service, 'get_external_user_id'):
+            user_id = user_service.get_external_user_id('lti')
```
Comment (taimoor-ahmed-1) [inline instruqtxblock/instruqtxblock.py:100]: empty line after conditional block

File: instruqtxblock/instruqtxblock.py [hunk 2]
```diff
@@ -89,17 +133,49 @@ def student_view(self, context=None):  # lint-amnesty, pylint: disable=unused-ar
...
         """
-        save_completion = False
-        try:
-            self.emit_completion(1.0)
-            save_completion = True
-        except BaseException as exp:
-            log.error("Error while marking completion %s", exp)
+        save_track_completion, save_challenge_completion = False, False
+
+        if data['event'] == "track.completed":
```
Comment (taimoor-ahmed-1) [inline instruqtxblock/instruqtxblock.py:141]: is there a chance `event` won't be in `data`? We can use `.get()` method to be safe.

**Review by madnihamza1841 — APPROVED** [review]
LGTM

File: instruqtxblock/instruqtxblock.py (hunk 2 above)
Comment (ziafazal) [reply instruqtxblock/instruqtxblock.py:141]: data must have `event` in case it does not we should raise exception`

**Review by taimoor-ahmed-1 — APPROVED** [review]
LGTM 👍

**Pull request merged**